weatherman.py

```diff
@@ -0,0 +1,192 @@
+import argparse
+import calendar
+from colorama import Fore, Style
+import csv
+import os
+
+def convert_month(month_input):
+    try:
+        month_num = int(month_input)
+        if 1 <= month_num <= 12:
+            return calendar.month_abbr[month_num]
+        else:
+            raise ValueError("Invalid month number")
+    except ValueError:
+        raise ValueError("Invalid input")
+    
+def read_files(get_year, get_month):
+    weather_details = []
+
+    base_dir = r"D:\Cogent Labs\weatherfiles\weatherfiles"
+    files_to_process = [filename for filename in os.listdir(base_dir)
+                        if get_year in filename and (not get_month or get_month in filename)]
+    
+    for filename in files_to_process:
+        file_path = os.path.join(base_dir, filename)
+
+        with open(file_path, 'r') as weather_file:
+            reader = csv.DictReader(weather_file)
+            weather_details.extend(reader)
+    
+    return weather_details
+
+def calculate_statistics(weather_details, args):
```

Comment:
calculate_statistics define notion to me, Please use meaning full name for every function